src/actions/accountsActions.js

```diff
@@ -238,27 +243,53 @@ export const setActiveAccountAction = (accountId: string) => {
   };
 };
 
-export const switchAccountAction = (accountId: string, privateKey?: string) => {
+/* eslint-disable */
+export const getPkAndInitSWSdkAction = (switchToAcc?: string) => {
+  return async (dispatch: Dispatch) => {
+    const currentScreenAndProps = getNavigationPathAndParamsState();
+    if (!currentScreenAndProps) return;
+    const { path, params } = currentScreenAndProps;
+    const returnToCurrentScreenAction = getActionForPathAndParams(path, params);
+    navigate(PIN_CODE, {
+      customUnlockAction: async (privateKey: string) => {
```

Comment:
Isn't it a bit weird to pass a function to the `navigate`? :thinking:

Reply:
Agree, does look weird, but cannot think of any other pragmatic way. Other approach would be to play with this in reducer and putting a flag that this action should be exectured?

Reply:
True, I've passed two variables now - one to check if PinUnlock screen action should open app or init Smart Wallet SDK and second one to check if it needed to switch to an account by doing so.